app.js

```diff
@@ -419,6 +419,44 @@ function ApiHandler(compileHandler) {
             }).join("\n"));
         }
     }, this));
+     // ---- this whole thing should be in a separate file etc, proof of concept only
+    // TODO: we'll need something smarter than this; URLs are long-lived and we nede
+    // to think about it; languages...formatters...etc.maybe something like:
+    var exec = require('./lib/exec');
+    this.handler.post('/format/clang-format-3.8', function (req, res) {
+        res.setHeader('Content-Type', 'application/json');
+        var style = "";
+        if (req.body.base && req.body.base !== "None") {
```

Comment:
I have a feeling that this logic can be simplifoed a bit. And maybe doing this is not the safest thing. That is a major concern